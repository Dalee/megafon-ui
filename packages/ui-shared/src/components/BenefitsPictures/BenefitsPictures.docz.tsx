// @ts-ignore
import img from './i/img.png';
// @ts-ignore
import imgShort from './i/img_short.png';

const item = (image) => {
    return {
        title: 'Lorem ipsum',
        text: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit',
        img: image,
    };
};

const getItems = (count, image) => new Array(count).fill(item(image));

export const twoItems = getItems(2, img);
export const threeItems = getItems(3, img);
export const fourItems = getItems(4, img);
export const twoItemsWithShortImg = getItems(2, imgShort);
