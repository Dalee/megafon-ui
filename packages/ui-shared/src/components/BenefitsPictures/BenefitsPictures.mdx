---
name: BenefitsPictures
menu: Constructor
route: components/benefits-pictures
---

import { Playground, Props } from 'docz';
import BenefitsPictures from './BenefitsPictures';
import { twoItems, threeItems, fourItems, twoItemsWithShortImg } from './BenefitsPictures.docz.tsx';
import { Grid, GridColumn } from '@megafon/ui-core';

# BenefitsPictures

## Properties
<Props of={BenefitsPictures} />

## Basic usage

### Left align, large grid gap
<Playground>
    <BenefitsPictures
        items={twoItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        items={threeItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        items={fourItems}
    />
</Playground>

### Left align, medium grid gap

<Playground>
    <BenefitsPictures
        gridGap="medium"
        items={twoItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        gridGap="medium"
        items={threeItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        gridGap="medium"
        items={fourItems}
    />
</Playground>

### Center align, large grid gap

<Playground>
    <BenefitsPictures
        hAlign="center"
        items={twoItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        hAlign="center"
        items={threeItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        hAlign="center"
        items={fourItems}
    />
</Playground>

### Center align, medium grid gap

<Playground>
    <BenefitsPictures
        gridGap="medium"
        hAlign="center"
        items={twoItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        gridGap="medium"
        hAlign="center"
        items={threeItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        gridGap="medium"
        hAlign="center"
        items={fourItems}
    />
</Playground>

### Centered image
<Playground>
    <BenefitsPictures
        items={twoItemsWithShortImg}
        hAlignImg="center"
    />
</Playground>

## Included props
``` javascript
const twoItems = [
    {
        title: 'Lorem ipsum',
        text: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit',
        img: image.png,
    },
    {
        title: 'Lorem ipsum',
        text: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit',
        img: image.png,
    },
]
```
