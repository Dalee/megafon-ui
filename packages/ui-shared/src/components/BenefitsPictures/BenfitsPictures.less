@import '~@megafon/ui-core/styles/base.less';

@b: e('.benefits-pictures');

@{b} {
    &__img {
        max-width: 100%;
        width: 100%;
        height: auto;
        margin-bottom: 32px;
    }

    &__img_h-align_center {
        display: block;
        width: auto;
        margin-left: auto;
        margin-right: auto;
    }

    &__title {
        margin-bottom: 12px;
    }
}
