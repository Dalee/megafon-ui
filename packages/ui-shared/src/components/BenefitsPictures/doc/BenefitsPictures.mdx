---
name: BenefitsPictures
menu: Конструктор
route: components/benefits-pictures
---

import Example from './BenefitsPictures.example.mdx';
import Props from './BenefitsPictures.props.mdx';

# BenefitsPictures

<ImportExample name="BenefitsPictures" />

<DoczTabs>
    <Example />
    <Props />
</DoczTabs>
