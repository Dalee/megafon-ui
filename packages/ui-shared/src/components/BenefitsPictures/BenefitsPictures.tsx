import * as React from 'react';
import throttle from 'lodash.throttle';
import './BenfitsPictures.less';
import { cnCreate, Grid, GridColumn, Header, Paragraph } from '@megafon/ui-core';
import { IBenefit, GridConfig, GridGutterSize } from './types';

const DESKTOP_MIDDLE_START = 1280;
const THROTTLE_TIME = 500;

export interface IBenefitsPictures {
    /** Benefits list */
    items: IBenefit[];
    /** Benefits horizontal align */
    hAlign?: 'left' | 'center';
    /** Grid gap size */
    gridGap: GridGutterSize;
    /** Image horizontal align */
    hAlignImg?: 'left' | 'center';
}

const columnSize: GridConfig = {
    wide: '4',
    desktop: '4',
    tablet: '5',
};

const isEvenIndex = index => !((index + 1) % 2);
const getEvenOffset = index =>  isEvenIndex(index) ? '1' : undefined;
const getOddOffset = index => isEvenIndex(index) ? undefined : '1';

const getLeftConfig = (count: number, index: number): GridConfig => {

    switch (count) {
        case 2:
        case 4:
            return {
                ...columnSize,
                leftOffsetWide: getEvenOffset(index),
                leftOffsetDesktop: getEvenOffset(index),
                leftOffsetTablet: getEvenOffset(index),
            };

        default:
            return {
                ...columnSize,
                leftOffsetTablet: getEvenOffset(index),
            };
    }
};

const getCenterConfig = (count: number, index: number, gutterSize: string ): GridConfig => {

    if (gutterSize === 'medium') {
        switch (count) {
            case 3:
                return {
                    ...columnSize,
                };

            case 4:
                return {
                    ...columnSize,
                    // wide
                    rightOffsetWide: getOddOffset(index),
                    leftOffsetWide: getEvenOffset(index),
                    // desktop
                    rightOffsetDesktop: getEvenOffset(index),
                    leftOffsetDesktop: getOddOffset(index),
                    // tablet
                    rightOffsetTablet: getEvenOffset(index),
                    leftOffsetTablet: getOddOffset(index),
                };

            default:
                return {
                    ...columnSize,
                    // wide
                    rightOffsetWide: getOddOffset(index),
                    leftOffsetWide: getEvenOffset(index),
                };
        }
    }

    switch (count) {
        case 4:
            return {
                ...columnSize,
                // wide
                leftOffsetWide: getOddOffset(index),
                rightOffsetWide: getEvenOffset(index),
                // desktop
                leftOffsetDesktop: getOddOffset(index),
                rightOffsetDesktop: getEvenOffset(index),
                // tablet
                leftOffsetTablet: getOddOffset(index),
                rightOffsetTablet: getEvenOffset(index),
            };

        case 3: {
            return {
                ...columnSize,
                // tablet
                leftOffsetTablet: getOddOffset(index),
            };
        }

        default:
            return {
                ...columnSize,
            };
    }
};

const cn = cnCreate('benefits-pictures');
const BenefitsPictures: React.FC<IBenefitsPictures> = ({
    items,
    hAlign = 'left',
    gridGap = 'large',
    hAlignImg = 'left',
}) => {
    const isLargeGutter = gridGap === 'large';
    const [currentGutter, setCurrentGutter] = React.useState(gridGap);

    const resizeHandler = () => {
        if (!isLargeGutter) {
            return;
        }

        if (window.innerWidth < DESKTOP_MIDDLE_START) {
            setCurrentGutter('medium');
        } else {
            setCurrentGutter('large');
        }
    };

    const ThrottledResizeHandler = throttle(resizeHandler, THROTTLE_TIME);

    React.useEffect(() => {
        resizeHandler();
        window.addEventListener('resize', ThrottledResizeHandler);

        return () => {
            window.removeEventListener('resize', ThrottledResizeHandler);
        };
    }, []);

    return (
        <div className={cn()}>
            <Grid
                guttersLeft={currentGutter}
                hAlign={hAlign === 'center' && items.length !== 3 ? 'center' : 'left'}
            >
                {items.map(({img, title, text}, index) =>
                    <GridColumn
                        {...hAlign === 'left'
                            ? getLeftConfig(items.length, index)
                            : getCenterConfig(items.length, index, gridGap)}
                        key={index}
                    >
                        <div className={cn('item')}>
                            <img className={cn('img', {
                                'h-align': hAlignImg,
                            })}
                                 src={img}
                                 alt=""
                            />
                            <Header className={cn('title')} hAlign={hAlign} as="h3">{title}</Header>
                            <Paragraph align={hAlign}>{text}</Paragraph>
                        </div>
                    </GridColumn>
                )}
            </Grid>
        </div>
    );
};

export default BenefitsPictures;
