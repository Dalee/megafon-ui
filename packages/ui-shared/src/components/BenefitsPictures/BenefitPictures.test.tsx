import * as React from 'react';
import { shallow } from 'enzyme';
import BenefitsPictures, { IBenefitsPictures } from './BenefitsPictures';
import { twoItems, threeItems, fourItems } from './BenefitsPictures.docz';

const props: IBenefitsPictures = {
    items: twoItems,
    hAlign: 'left',
    gridGap: 'large',
    hAlignImg: 'left',
};

const itemsSet = [
    twoItems,
    threeItems,
    fourItems,
];

describe('<BenefitsPictures />', () => {
    it('renders BenefitsPictures', () => {
        const wrapper = shallow(<BenefitsPictures {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render with center horizontal align', () => {
        itemsSet.forEach(items => {
                const wrapper = shallow(<BenefitsPictures {...props} items={items} hAlign="center"/>);
                expect(wrapper).toMatchSnapshot();
        });
    });

    it('render with medium grid gap', () => {
        itemsSet.forEach(items => {
            const wrapper = shallow(<BenefitsPictures {...props} items={items} gridGap="medium" />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    it('render with center image align', () => {
        itemsSet.forEach(items => {
            const wrapper = shallow(<BenefitsPictures {...props} items={items} hAlignImg="center" />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
