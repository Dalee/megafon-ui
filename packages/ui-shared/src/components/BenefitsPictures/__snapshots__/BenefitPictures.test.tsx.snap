// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BenefitsPictures /> render with center horizontal align 1`] = `
<div
  className="benefits-pictures"
>
  <Grid
    guttersLeft="large"
    hAlign="center"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_center"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="center"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="center"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_center"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="center"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="center"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> render with center horizontal align 2`] = `
<div
  className="benefits-pictures"
>
  <Grid
    guttersLeft="large"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      leftOffsetTablet="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_center"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="center"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="center"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_center"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="center"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="center"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="2"
      leftOffsetTablet="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_center"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="center"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="center"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> render with center horizontal align 3`] = `
<div
  className="benefits-pictures"
>
  <Grid
    guttersLeft="large"
    hAlign="center"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_center"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="center"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="center"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      rightOffsetDesktop="1"
      rightOffsetTablet="1"
      rightOffsetWide="1"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_center"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="center"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="center"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="2"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_center"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="center"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="center"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="3"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      rightOffsetDesktop="1"
      rightOffsetTablet="1"
      rightOffsetWide="1"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_center"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="center"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="center"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> render with medium grid gap 1`] = `
<div
  className="benefits-pictures"
>
  <Grid
    guttersLeft="medium"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> render with medium grid gap 2`] = `
<div
  className="benefits-pictures"
>
  <Grid
    guttersLeft="medium"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetTablet="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="2"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> render with medium grid gap 3`] = `
<div
  className="benefits-pictures"
>
  <Grid
    guttersLeft="medium"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="2"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="3"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> renders BenefitsPictures 1`] = `
<div
  className="benefits-pictures"
>
  <Grid
    guttersLeft="large"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="benefits-pictures__item"
      >
        <img
          alt=""
          className="benefits-pictures__img benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;
